File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    t = num
    cnt = 0
    while True:
        if t == 1:
            return cnt
        elif cnt == 500:
            return -1
        if t % 2 == 0:
            t /= 2
        else:
            t = t * 3 + 1
        cnt += 1

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        long target = num;
        
        for (int i = 0; i < 500; i++) {
            if (target == 1) return answer;
            
            if (target % 2 == 0) {
                target /= 2;
            } else {
                target = target * 3 + 1;
            }
            
            answer++;
        }
        
        return -1;
    }
}